core/src/main/java/com/rudderstack/android/sdk/core/util/Utils.java

```diff
@@ -269,11 +269,57 @@ public static String appendSlashToUrl(@NonNull String dataPlaneUrl) {
         return dataPlaneUrl;
     }
 
-    public static boolean isEmpty(@Nullable Map<String, String> value) {
+    public static boolean isEmpty(@Nullable Map value) {
         return (value == null || value.isEmpty());
     }
 
     public static boolean isEmpty(@Nullable String value) {
         return (value == null || value.isEmpty());
     }
+
+    public static boolean isEmpty(@Nullable List value) {
+        return (value == null || value.isEmpty());
+    }
+
+    public static String getString(@Nullable Object value) {
+        if (value == null) {
+            return null;
+        }
+        switch (getType(value)) {
+            case "Byte":
+            case "Short":
+            case "Integer":
+            case "Long":
+            case "Float":
+            case "Double":
+            case "Boolean":
+            case "Character":
+            case "ArrayList":
+            case "HashMap":
+                return value.toString();
+            case "String":
+                return (String) value;
+            case "Array":
+                return new Gson().toJson(value);
+            default:
+                return null;
+        }
+    }
+
+    public static boolean getBoolean(@Nullable Object value) {
+        if (value == null) {
+            return false;
+        }
+        switch (getType(value)) {
+            case "Boolean": return (boolean) value;
+            default: return false;
+        }
+    }
+
+    private static String getType(Object object) {
+        if (object.getClass().isArray()) {
+            return "Array";
+        }
+        return object.getClass().getSimpleName();
+    }
```

Comment:
I guess we can remove them if we go ahead with the Gson approach for constructing the `ResidencyUrl` objects